File: src/main/java/wraith/fabricaeexnihilo/api/recipes/barrel/CompostRecipe.java
package wraith.fabricaeexnihilo.api.recipes.barrel;

import net.minecraft.item.ItemConvertible;
import net.minecraft.item.ItemStack;
import wraith.fabricaeexnihilo.api.crafting.ItemIngredient;
import wraith.fabricaeexnihilo.util.Color;

import java.util.function.Predicate;

public record CompostRecipe(ItemIngredient ingredient,
                            ItemStack result,
                            double amount,
                            Color color) implements Predicate<ItemStack> {

    @Override
    public boolean test(ItemStack stack) {
        return ingredient.test(stack);
    }

    public void test(ItemConvertible stack) {
        ingredient.test(stack);
    }

}

File: src/main/kotlin/exnihilofabrico/api/recipes/CompostRecipe.kt
package exnihilofabrico.api.recipes

import exnihilofabrico.api.crafting.ItemIngredient
import exnihilofabrico.util.Color
import net.minecraft.item.ItemConvertible
import net.minecraft.item.ItemStack
import java.util.function.Predicate

data class CompostRecipe(val ingredient: ItemIngredient, val result: ItemStack, val amount: Double, val color: Color): Predicate<ItemStack> {
    override fun test(t: ItemStack) = ingredient.test(t)
    fun test(t: ItemConvertible) = ingredient.test(t)
}